sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1873,6 +1874,22 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val RECURSION_LEVEL_LIMIT = buildConf("spark.sql.cte.recursion.level.limit")
+    .internal()
+    .doc("Maximum level of recursion that is allowed wile executing a recursive CTE definition." +
+      "If a query does not get exhausted before reaching this limit it fails.")
+    .intConf
+    .createWithDefault(100)
```

Comment:
Can you use "-1" for the unlimited case?

Reply:
Good idea. Done.